### PR TITLE
Provider API refactor

### PR DESCRIPTION
**Pull request by Thurfjell**
### Ticket:

- [19955](https://dev.azure.com/OGCIO-Digital-Services/Digital%20Services%20Programme/_workitems/edit/19955)

### Description

Refactor all provider API endpoints (sms and email) into a `/providers` resource.
Removed all old references.

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [ ] **New feature**
- [x] **Dev change**

### Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This change might impact the developer experience of others and should be communicated

### Screenshots:

N/A

### CODE REVIEW

File: apps/messaging-api/routes/providers/index.ts [hunk 1]
```diff
@@ -0,0 +1,722 @@
...
+      const totalCount = query.rows.at(0)?.count || 0;
+      const url = new URL(`/api/v1/${prefix}`, process.env.HOST_URL).href;
+      const links = getPaginationLinks({
+        totalCount,
+        url,
+        limit,
+        offset,
+      });
+
+      const response: GenericResponseSingle<Static<typeof ProviderList>> = {
```
Comment (SamSalvatico) [inline apps/messaging-api/routes/providers/index.ts:220]: Why using "Single" type given that the response is a list?

File: apps/messaging/app/[locale]/admin/providers/SmsProviders.tsx [hunk 2]
```diff
@@ -14,16 +14,14 @@ export default async () => {
 
   const { data: providers } = await sdk.getSmsProviders();
 
+  console.log(providers);
```
Comment (SamSalvatico) [inline apps/messaging/app/[locale]/admin/providers/SmsProviders.tsx:17]:
```suggestion

```

File: apps/messaging-api/routes/providers/index.ts (hunk 1 above)
Comment (Thurfjell) [reply apps/messaging-api/routes/providers/index.ts:220]: Aha I think we should actually get rid of the one setting the generic type to an array. That's definitely something that the implementor should have to decide imo.

Some bad naming.

Comment (SamSalvatico) [reply apps/messaging-api/routes/providers/index.ts:220]: agree

**Pull request merged**